docs/ValidationRules/RULEID.RULEFRIENDLYNAME.cs

```diff
@@ -0,0 +1,142 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+
+/*
+ * INSTRUCTIONS:
+ * 
+ * - Replace `RULEID` with a valid value.
+ *      It should be start with a prefix "SARIF", followed by 
+ *      a 4 digit number which is the next sequential number available
+ *      to use in ~\Sarif.Multitool\Rules\ directory.
+ *      Example:
+ *          SARIF1023
+ *
+ * - Replace `RULEFRIENDLYNAME` with a valid value.
+ *      RULEFRIENDLYNAME should concisely define the purpose of this rule.
+ *      Where possible, prefer using Imperative language, like `UseAbsoluteUri`
+ *      instead of Indicative language, like `UriIsNotAbsolute`.
+ *      Examples:
+ *          DoNotUseFriendlyNameAsRuleId
+ *          ReferToFinalSchema
+ *
+ * - Rename this file as <RULEID>.<RULEFRIENDLYNAME>.cs.
+ * 
+ * - Remove All INSTRUCTIONS after the changes have been made.
+ */
+namespace Microsoft.CodeAnalysis.Sarif.Multitool.Rules
+{
+    public class RULEFRIENDLYNAME : SarifValidationSkimmerBase
+    {
+        private readonly MultiformatMessageString _fullDescription = new MultiformatMessageString
+        {
+            /*
+             * INSTRUCTIONS:
+             * Add a new key-value pair in ~\Sarif.Multitool\Rules\RuleResources.resx.
+             * 
+             * Key: 
+             *      RULEID_RULEFRIENDLYNAME
+             *      
+             * Value: 
+             *      Provide a brief description with atleast two sentences, both ending in a period.
+             *      The first sentence should be a short description of the rule.
+             *      The second sentence must describe the utility/usage of the rule.
+             *      
+             * Example:
+             *      The $schema property should be present, and must refer to the final version 
+             *      of the SARIF 2.1.0 schema. This enables IDEs to provide Intellisense for SARIF log files.
+             *      
+             * Notes:
+             *      The first sentence will be used as a `ShortDescription` for this rule.
+             *      All sentences together will be used as a `LongDescription` for this rule.
+             */
+            Text = RuleResources.RULEID_RULEFRIENDLYNAME_Description
+        };
+
+        public override MultiformatMessageString FullDescription => _fullDescription;
```

Comment:
>_fullDescription [](start = 68, length = 16)

We don't need the `_fullDescription` field. Just this:

```
public override MultiformatMessageString FullDescription = new MultiformatMessageString
{
    Text = RuleResources.RULEID_RULEFRIENDLYNAME
}
```

Reply:
hah! look at iteration 1,i had original put a todo here to remind myself to ask u if we should remove it! and then silli-ly (is that a word?) removed the todo while i was making further edits :D

---
In reply to: [435558588](https://github.com/microsoft/sarif-sdk/pull/1902#discussion_r435558588) [](ancestors = 435558588)